src/services/earn/earn-service.ts

```diff
@@ -475,6 +476,103 @@ export class EarnService implements IEarnService {
     return buildCompanionMulticall({ chainId, calls, value: increaseInfo.value });
   }
 
+  async buildClaimDelayedWithdrawPositionTx({
+    chainId,
+    positionId,
+    recipient,
+    permissionPermit,
+    claim,
+  }: {
+    chainId: ChainId;
+    positionId: PositionId;
+    recipient: Address;
+    permissionPermit?: EarnPermissionPermit;
+    claim: { tokens: { token: TokenAddress; convertTo?: TokenAddress }[]; swapConfig?: EarnActionSwapConfig };
+  }): Promise<BuiltTransaction> {
+    const vault = EARN_VAULT.address(chainId);
+    const manager = DELAYED_WITHDRAWAL_MANAGER.address(chainId);
+    const [, , tokenId] = positionId.split('-');
+    const bigIntPositionId = BigInt(tokenId);
+    const claimsToConvert = claim.tokens.filter(({ convertTo }) => !!convertTo);
+
+    if (claimsToConvert.length == 0) {
```

Comment:
I think we might need to re-structure this a little bit. If `claimsToConvert.length > 0` but `withdrawableFunds` for that token in 0, then we should also just call the manager. If `claimsToConvert.length > 0`, then we could calculate `withdrawableFunds` here and then re-filter `claimsToConvert` (under a new name or not)